spring-cloud-scheduler-spi-test/src/main/java/org/springframework/cloud/scheduler/spi/test/AbstractIntegrationTests.java

```diff
@@ -0,0 +1,461 @@
+		Timeout timeout = scheduleTimeout();
+		ScheduleInfo scheduleInfo = new ScheduleInfo();
+		scheduleInfo.setScheduleName(scheduleName+0);
+		scheduleInfo.setTaskDefinitionName(definitionName+0);
+		assertThat(scheduleInfo, eventually(hasSpecifiedSchedulesByTaskDefinitionName(taskScheduler().list(definitionName+0), scheduleInfo.getTaskDefinitionName(), 2), timeout.maxAttempts, timeout.pause));
+	}
+
+	private ScheduleInfo createAndVerifySchedule() {
+		ScheduleRequest request = createScheduleRequest();
+		schedule(request, request.getDefinition());
+		ScheduleInfo scheduleInfo = new ScheduleInfo();
+		scheduleInfo.setScheduleName(request.getScheduleName());
+		verifySchedule(scheduleInfo);
+		return scheduleInfo;
+	}
+
+	private ScheduleRequest createScheduleRequest() {
+		String definitionName = randomName();
+		String scheduleName = "ScheduleName_" + definitionName;
+		return createScheduleRequest(scheduleName, definitionName);
+	}
+
+	private ScheduleRequest createScheduleRequest(String scheduleName, String definitionName) {
+		AppDefinition definition = new AppDefinition(definitionName, testAppProperties());
+		return new ScheduleRequest(definition, testSchedulerProperties(), testDeploymentProperties(), testCommandLineArgs(), scheduleName, testApplication());
+	}
+
+	private void verifySchedule(ScheduleInfo scheduleInfo) {
+		Timeout timeout = scheduleTimeout();
+		assertThat(scheduleInfo, eventually(hasSpecifiedSchedule(taskScheduler().list(), scheduleInfo.getScheduleName()), timeout.maxAttempts, timeout.pause));
+	}
+
+	private  void schedule(ScheduleRequest request, AppDefinition appDefinition) {
+		log.info("Scheduling {}...", request.getScheduleName());
+		taskScheduler().schedule(request);
+
+	}
+
+	private void unschedule(String scheduleName) {
+		log.info("unscheduling {}...", scheduleName);
+
+		Timeout timeout = unscheduleTimeout();
+		taskScheduler().unschedule(scheduleName);
+
+		ScheduleInfo scheduleInfo = new ScheduleInfo();
+		scheduleInfo.setScheduleName(scheduleName);
+		assertThat(scheduleInfo, eventually(specifiedScheduleNotPresent(taskScheduler().list(), scheduleName), timeout.maxAttempts, timeout.pause));
+
+	}
+
+	protected String randomName() {
+		return name.getMethodName() + "-" + UUID.randomUUID().toString();
+	}
+
+	/**
+	 * Return the timeout to use for repeatedly querying that a task has been scheduled.
+	 * Default value is one minute, being queried every 5 seconds.
+	 */
+	protected Timeout scheduleTimeout() {
+		return new Timeout(12, 5000);
+	}
+
+	/**
+	 * Return the timeout to use for repeatedly querying whether a task has been unscheduled.
+	 * Default value is one minute, being queried every 5 seconds.
+	 */
+	protected Timeout unscheduleTimeout() {
+		return new Timeout(12, 5000);
+	}
+
+	/**
+	 * Return the time to wait between reusing scheduler requests. This could be necessary to give
+	 * some platforms time to clean up after unscheduling.
+	 */
+	protected int redeploymentPause() {
+		return 0;
+	}
```

Comment:
Why are these three hard coded within methods?  This feels like a crude way to provide defaults that are supposed to be overridable.  Why not just have class variables and exposed setters (so I don't have to override a method to  override a default)?

Reply:
done